### PR TITLE
Change Credential Google auth parameter

### PR DESCRIPTION
**Pull request by ericjordanmossman**
This pull request intends to resolve confusion around the intended use of google auth credentials. Ultimately `Credential` in RealmSwift should be an associated enum, but this is the best immediate change.

Addresses 
https://jira.mongodb.org/browse/RCOCOA-104
https://jira.mongodb.org/browse/RCOCOA-871

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,13 @@ x.y.z Release notes (yyyy-MM-dd)
 * <How to hit and notice issue? what was the impact?> ([#????](https://github.com/realm/realm-cocoa/issues/????), since v?.?.?)
 * None.
 
-<!-- ### Breaking Changes - ONLY INCLUDE FOR NEW MAJOR version -->
+### Breaking Changes
+* The following functionality has changed for accuracy:
+
+| Old API                                                      | New API                                                        |
+|:-------------------------------------------------------------|:---------------------------------------------------------------|
+| `Credentials(googleToken:)`                                           | `Credentials(googleAuthCode:)`                             |
```
Comment (jsflax) [inline CHANGELOG.md:15]: This seems misaligned. Also this table seems excessive for two small changes:
```md
* Change Google Credential parameter names to better reflect the required auth code:
    * `Credentials(googleToken:)` => `Credentials(googleAuthCode:)`
    * `[RLMCredentials credentialsWithGoogleToken:]` => `[RLMCredentials credentialsWithGoogleAuthCode:]`
```
Remember as well that CHANGELOG additions should be in the imperative mood (we don't do this everywhere, but we should).

**Review by jsflax — APPROVED** [review]
One change for CHANGELOG, otherwise LGTM 👍

**Comment by ericjordanmossman** [comment]
[osx Debug 11.5](https://ci.realm.io/job/objc_pr/configuration=Debug,target=osx,xcode_version=11.5/7129/console): "No Xcode found with version 11.5"
[osx Debug 11.6](https://ci.realm.io/job/objc_pr/configuration=Debug,target=osx,xcode_version=11.6/7129/console): "No Xcode found with version 11.6"
[osx Release 11.5](https://ci.realm.io/job/objc_pr/configuration=Release,target=osx,xcode_version=11.5/7129/console): "No Xcode found with version 11.5"
[osx Release 11.6](https://ci.realm.io/job/objc_pr/configuration=Release,target=osx,xcode_version=11.6/7129/console): "No Xcode found with version 11.6"
[osx-swift Debug 11.5](https://ci.realm.io/job/objc_pr/configuration=Debug,target=osx-swift,xcode_version=11.5/7129/console): "No Xcode found with version 11.5"
[osx-swift Debug 11.6](https://ci.realm.io/job/objc_pr/configuration=Debug,target=osx-swift,xcode_version=11.6/7129/console): "No Xcode found with version 11.6"
[osx-swift Release 11.5](https://ci.realm.io/job/objc_pr/configuration=Release,target=osx-swift,xcode_version=11.5/7129/): "No Xcode found with version 11.5"
[osx-swift Release 11.6](https://ci.realm.io/job/objc_pr/configuration=Release,target=osx-swift,xcode_version=11.6/7129/console): "No Xcode found with version 11.6"
[Jenkins: ios-static Release 12.0](https://ci.realm.io/job/objc_pr/configuration=Release,target=ios-static,xcode_version=12.0/7129/console): 
"
Build step 'Execute shell' marked build as failure
ERROR: Step ‘Set GitHub commit status (universal)’ failed: no workspace for objc_pr/configuration=Release,target=ios-static,xcode_version=12.0 #7129
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
ERROR: Step ‘Delete workspace when build is done’ failed: no workspace for objc_pr/configuration=Release,target=ios-static,xcode_version=12.0 #7129
Finished: FAILURE
"
@jsflax All the 11.5/11.6 portions fail with no Xcode found. I imagine this points to CI? Is this good to squash and merge?

**Comment by jsflax** [comment]
Yes, but seems like @tgoyne should take a look at that comment? Those are some strange failures.

**Comment by tgoyne** [comment]
I removed Xcode 11.5 and 11.6 from the CI machines a while ago (because 11.5-11.7 all have the same compiler versions) but the change to remove them from PR testing hasn't made it to v10 yet.

**Pull request merged**